### ISSUE TITLE
Cannot launch gtk apps from alacritty

### ISSUE THREAD

**Issue by milesfrain** [alacritty/alacritty#5613]
I'm not able to launch `meld` from within an `alacritty` terminal, but I can launch it from the standard ubuntu terminal.

Interestingly, if I launch meld from a standard terminal first and leave it running, I can then launch another meld instance successfully from the alacritty terminal. But if I then close both instances, I get the same error when trying to launch meld again from alacritty.

Another workaround is to unset the `GDK_PIXBUF_*` vars, but I don't know what else that might break:
```
unset GDK_PIXBUF_MODULEDIR
unset GDK_PIXBUF_MODULE_FILE
```
https://askubuntu.com/questions/1351636/gvim-failed-to-load-usr-share-icons-yaru-16x16-status-image-missing-png-unrec

### Logs

```
meld
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
[1]    4503 abort (core dumped)  meld
```

```
env | grep GDK_PIXBUF
GDK_PIXBUF_MODULEDIR=/snap/alacritty/46/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders
GDK_PIXBUF_MODULE_FILE=/home/miles/snap/alacritty/common/.cache/gdk-pixbuf-loaders.cache
```

### System

Ubuntu 20.04
```
alacritty --version
alacritty 0.8.0 (a1b13e6)
```

**Comment by chrisduerr** [comment]
First of all I'm not sure why you have `GDK_PIXBUF` environment variables exported at all, but making sure all environment variables are the same for both terminals should be the first step. It's really none of Alacritty's business to make sure environment variables are set to anything specific.

Ultimately this really isn't Alacritty's problem. Things like Wayland/XWayland might be inherited *through* Alacritty, but that's really still mostly about users screwing with their system not Alacritty doing anything surprising.

**Comment by YanzhaoW** [comment]
I had exact the same problem. In my case, I installed alacritty through snap and somehow `GDK_PIXBUF_*` vars are set through snap to the folder `/snap/alacritty`. You can check how the env vars are set by 
```
PS4='+$BASH_SOURCE> ' BASH_XTRACEFD=7 bash -xl 7>&2
```
There you can search the word "alacritty".

Well, my solution is deleting alacritty from snap and reinstall it by source or cargo.

tldr: don't use snap

**Comment by kevclark** [comment]
I also have this issue on U 20.04, with Alacritty installed via snap. `GDK_PIXBUF_` are not normally set on my system, only when launching via snapd.  Like the OP I just unset these via alacritty.yaml then launching my the shell again as a subshell.  Works for me.
```
shell:
 program: /usr/bin/zsh
 # These GDK vars are being set by the snap installed alacritty which 
 # stop GTK programs from launching from the cmdline
 args:
   - -c
   - unset GDK_PIXBUF_MODULEDIR && unset GDK_PIXBUF_MODULE_FILE && exec zsh
```
